File: php/64-minimum-path-sum.php
<?php
class Solution {
    /**
     * @param Integer[][] $grid
     * @return Integer
     */
    function minPathSum2($grid) {
        $row = count($grid);
        $col = count($grid[0]);
        foreach ($grid as $i => $r) {
            foreach ($r as $j => $c) {
                $min = min($grid[$i-1][$j]?? PHP_INT_MAX, $grid[$i][$j-1]?? PHP_INT_MAX);
                $grid[$i][$j] += $min == PHP_INT_MAX? 0 : $min;
            }
        }
        return $grid[$row-1][$col-1];
    }
}

File: go/64-minimum-path-sum.go
package main

func minPathSum(grid [][]int) int {
	row := len(grid)
	col := len(grid[0])
	max := 1 << 32
	for rk, row := range grid {
		for ck, _ := range row {
			up := max
			if rk-1 >= 0 {
				up = grid[rk-1][ck]
			}
			left := max
			if ck-1 >= 0 {
				left = grid[rk][ck-1]
			}
			min := int(math.Min(float64(left), float64(up)))
			if min == max {
				min = 0
			}
			grid[rk][ck] += min
		}
	}

	return grid[row-1][col-1]
}
